src/scripts/main.js

```diff
@@ -1 +1,80 @@
 'use strict';
+
+const firstPromise = () => {
+  return new Promise((resolve, reject) => {
+    document.body.addEventListener('click', () => {
+      resolve('First promise was resolved');
+    });
+
+    setTimeout(() => {
+      reject(new Error('First promise was rejected'));
+    }, 3000);
+  });
+};
+
+const secondPromise = () => {
+  return new Promise((resolve) => {
+    document.body.addEventListener('click', () => {
+      resolve('Second promise was resolved');
+    });
+
+    document.body.addEventListener('contextmenu', (e) => {
+      e.preventDefault();
+      resolve('Second promise was resolved');
```

Comment:
This line resolves the second promise on a right-click, but it does not prevent the promise from resolving again on a subsequent left-click. Consider adding logic to ensure the promise resolves only once.